broadinstitute/single_cell_portal_core
# Removing auth requirement for XDSS, retiring feature flags (SCP-4163)

This update removes the requirement for users to be signed in for cross-dataset search requests.  In addition, the `cross_dataset_search_[frontend|backend]` feature flags have both been retired.

MANUAL TESTING
1. Pull this branch and run migrations with `bin/rails db:migrate`
2. In the Rails console, validate that both XDSS feature flags are gone:
```
irb(main):003:0> FeatureFlag.where(name: /cross_dataset_search/).count
=> 0
```
3. Boot as normal, and confirm that facet filters include results from Azul without needing to sign in
4. Run a search for `species:Homo sapiens` and `disease:HIV infectious disease` and confirm the HCA projects `An Atlas of Immune Cell Exhaustion in HIV-Infected Individuals Revealed by Single-Cell Transcriptomics` and `Single-Cell RNA-Seq Reveals Transcriptional Heterogeneity in Latent and Reactivated HIV-infected Cells` are returned
5. (Optional) sign in and re-run the search, confirming that the HCA results are the same

This PR satisfies SCP-4163.
